projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.ts

```diff
@@ -75,18 +139,81 @@ export class CartCouponComponent implements OnInit, OnDestroy {
           this.onSuccess(success);
         })
     );
+
+    this.subscription.add(
+      this.cartVoucherService.getAddVoucherResultError().subscribe(error => {
+        this.onError(error);
+      })
+    );
+  }
+
+  protected onError(error: boolean) {
+    if (error) {
+      this.customerCouponService.loadCustomerCoupons(
+        this.MAX_CUSTOMER_COUPON_PAGE
+      );
+      this.cartVoucherService.resetAddVoucherProcessingState();
+    }
   }
 
-  onSuccess(success: boolean) {
+  protected onSuccess(success: boolean) {
```

Comment:
please remove `protected` from here, since it's introducing potentional BC. leave it as it was, simply `onSuccess()`

Reply:
updated.